Source/objects.cpp

```diff
@@ -5282,37 +5282,37 @@ void SyncObjectAnim(int o)
 
 	const int i = std::distance(std::begin(ObjFileList), found);
 
-	Objects[o]._oAnimData = pObjCels[i].get();
-	switch (Objects[o]._otype) {
+	object._oAnimData = pObjCels[i].get();
+	switch (object._otype) {
 	case OBJ_L1LDOOR:
 	case OBJ_L1RDOOR:
-		SyncL1Doors(Objects[o]);
+		SyncL1Doors(object);
 		break;
 	case OBJ_L2LDOOR:
 	case OBJ_L2RDOOR:
-		SyncL2Doors(Objects[o]);
+		SyncL2Doors(object);
 		break;
 	case OBJ_L3LDOOR:
 	case OBJ_L3RDOOR:
-		SyncL3Doors(Objects[o]);
+		SyncL3Doors(object);
 		break;
 	case OBJ_CRUX1:
 	case OBJ_CRUX2:
 	case OBJ_CRUX3:
-		SyncCrux(o);
+		SyncCrux(object);
 		break;
 	case OBJ_LEVER:
 	case OBJ_BOOK2L:
 	case OBJ_SWITCHSKL:
-		SyncLever(o);
+		SyncLever(object);
 		break;
 	case OBJ_BOOK2R:
 	case OBJ_BLINDBOOK:
 	case OBJ_STEELTOME:
-		SyncQSTLever(o);
+		SyncQSTLever(object);
 		break;
 	case OBJ_PEDISTAL:
-		SyncPedistal(o);
+		SyncPedestal(object, { setpc_x, setpc_y }, setpc_w);

```

Comment:
Passing the other things here is a bit out of scope of the PR I think. Not that I don't think its a good idea, though.